### ISSUE TITLE
How do we calculate performances on the test set ?

### ISSUE THREAD

**Issue by amaralibey** [mapillary/mapillary_sls#15]
Sorry to ask this, but is there any way we can calculate performances on the test set ? I don't see any "postprocessed.csv" in the test directories.
Is there a website where I can submit the results ? There arepeople talking about a challenge in the other discussions.

Thanks

**Comment by FrederikWarburg** [comment]
Hi!

We hosted a challenge as part of the visual localisation workshop for ECCV2020. 
I believe that the evaluation server is still functional. You'll need to upload a zip file with a csv file that contains query IDs and IDs for top N predictions. Here is a link to the evaluation server: https://codalab.mapillary.com/competitions/45#learn_the_details

For more details on the format of the csv file, please checkout the example files: https://github.com/mapillary/mapillary_sls/tree/master/files and the function https://github.com/mapillary/mapillary_sls/blob/b9edb130d3c8c248df79693fe8bfe138ad9f2ef3/mapillary_sls/utils/eval.py#L96

Hope it works out :)

**Comment by mlopezantequera** [comment]
The challenge is closed, so the evaluation server does not run anymore. We hope to re-open it for future workshops and hopefully leave it open as a running benchmark.

**Comment by dinarkino** [comment]
> The challenge is closed, so the evaluation server does not run anymore. We hope to re-open it for future workshops and hopefully leave it open as a running benchmark.

Hi, thank you for your work! 

Is it possible to publish test ground truth or make the evaluation server alive? I'm writing the paper, and I planned to use your dataset for training and as a benchmark. At the moment, it seems that I can only use the validation set for comparison, but even in your paper, you published only results on the test part of the dataset. The availability of the test ground truth will be much appreciated.

**Comment by Nanxiaohu666** [comment]
> Hi!
> 
> We hosted a challenge as part of the visual localisation workshop for ECCV2020. I believe that the evaluation server is still functional. You'll need to upload a zip file with a csv file that contains query IDs and IDs for top N predictions. Here is a link to the evaluation server: https://codalab.mapillary.com/competitions/45#learn_the_details
> 
> For more details on the format of the csv file, please checkout the example files: https://github.com/mapillary/mapillary_sls/tree/master/files and the function
> 
> https://github.com/mapillary/mapillary_sls/blob/b9edb130d3c8c248df79693fe8bfe138ad9f2ef3/mapillary_sls/utils/eval.py#L96
> 
> Hope it works out :)
Hello. I participate the MSLS Place recognition challenge. I want to know how to generate the .csv file used in eval?  In details. What is N number in topN results. Looking forward to your reply

**Comment by mlopezantequera** [comment]
We've re-opened the evaluation server as part of the [4th Long-Term Visual Localization under Changing Conditions
](https://sites.google.com/view/ltvl2021/), available [here](https://competitions.codalab.org/competitions/34623).

After the workshop challenge submission date is over (Oct 7th), we plan to re-open an ongoing evaluation server soon after.

**Comment by mlopezantequera** [comment]
The evaluation server is now permanently open. Read more here:
https://github.com/mapillary/mapillary_sls/#-evaluate-on-the-test-set
